plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/mutate/MutateProcessor.java

```diff
@@ -0,0 +1,305 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.mutate;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.processor.ConfigurationUtils;
+import org.elasticsearch.ingest.processor.Processor;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public final class MutateProcessor implements Processor {
+
+    public static final String TYPE = "mutate";
+
+    private final Map<String, Object> update;
+    private final Map<String, String> rename;
+    private final Map<String, String> convert;
+    private final Map<String, String> split;
+    private final Map<String, Tuple<Pattern, String>> gsub;
+    private final Map<String, String> join;
+    private final List<String> remove;
+    private final List<String> trim;
+    private final List<String> uppercase;
+    private final List<String> lowercase;
+
+    public MutateProcessor(Map<String, Object> update,
+                           Map<String, String> rename,
+                           Map<String, String> convert,
+                           Map<String, String> split,
+                           Map<String, Tuple<Pattern, String>> gsub,
+                           Map<String, String> join,
+                           List<String> remove,
+                           List<String> trim,
+                           List<String> uppercase,
+                           List<String> lowercase) {
+        this.update = update;
+        this.rename = rename;
+        this.convert = convert;
+        this.split = split;
+        this.gsub = gsub;
+        this.join = join;
+        this.remove = remove;
+        this.trim = trim;
+        this.uppercase = uppercase;
+        this.lowercase = lowercase;
+    }
+
+    public Map<String, Object> getUpdate() {
+        return update;
+    }
+
+    public Map<String, String> getRename() {
+        return rename;
+    }
+
+    public Map<String, String> getConvert() {
+        return convert;
+    }
+
+    public Map<String, String> getSplit() {
+        return split;
+    }
+
+    public Map<String, Tuple<Pattern, String>> getGsub() {
+        return gsub;
+    }
+
+    public Map<String, String> getJoin() {
+        return join;
+    }
+
+    public List<String> getRemove() {
+        return remove;
+    }
+
+    public List<String> getTrim() {
+        return trim;
+    }
+
+    public List<String> getUppercase() {
+        return uppercase;
+    }
+
+    public List<String> getLowercase() {
+        return lowercase;
+    }
+
+    @Override
+    public void execute(Data data) {
+        if (update != null) { doUpdate(data); }
```

Comment:
I think for consistency with the rest of the code base we should use three lines for these if statements?